src/test/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/sentencing/SentencingResourceIntTest.kt

```diff
@@ -1959,22 +1991,251 @@ class SentencingResourceIntTest : IntegrationTestBase() {
+                }
+              }
+              release()
+            }.bookingId
+            booking {
+              courtCase(
+                reportingStaff = staff,
+                beginDate = LocalDate.parse("2020-01-01"),
+                statusUpdateDate = LocalDate.parse("2020-01-01"),
+                statusUpdateStaff = staff,
+              ) {
+                val charge = offenderCharge(offenceCode = "HP09006", plea = "NG", resultCode1 = "2006")
+                courtEvent(eventDateTime = LocalDateTime.parse("2020-01-01T10:00"), outcomeReasonCode = "2006") {
+                  courtEventCharge(
+                    resultCode1 = "2006",
+                    offenderCharge = charge,
+                  )
+                }
+              }
+              release()
+            }
+          }
+        }
+      }
+
+      @Test
+      internal fun `cases copied are returned`() {
+        val response: BookingCourtCaseCloneResponse = webTestClient.post().uri("/prisoners/booking-id/$previousBookingId/sentencing/court-cases/clone")
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_SENTENCING")))
+          .exchange()
+          .expectBodyResponse()
+
+        assertThat(response.courtCases).hasSize(2)
+      }
+
+      @Test
+      internal fun `cases are merged into the booking along with existing cases`() {
+        transaction {
+          assertThat(offenderBookingRepository.findByIdOrNull(latestBookingId)!!.courtCases).hasSize(1)
+        }
+
+        webTestClient.post().uri("/prisoners/booking-id/$previousBookingId/sentencing/court-cases/clone")
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_SENTENCING")))
+          .exchange()
+          .expectStatus().isOk
+
+        transaction {
+          assertThat(offenderBookingRepository.findByIdOrNull(latestBookingId)!!.courtCases).hasSize(3)
+        }
+      }
+
+      @Test
+      internal fun `core cases details copied are copied`() {
+        webTestClient.post().uri("/prisoners/booking-id/$previousBookingId/sentencing/court-cases/clone")
+          .headers(setAuthorisation(roles = listOf("ROLE_NOMIS_SENTENCING")))
+          .exchange()
+          .expectStatus().isOk
+
+        transaction {
+          with(offenderBookingRepository.findByIdOrNull(latestBookingId)!!.courtCases.find { it.primaryCaseInfoNumber == "X0001" }!!) {
+            assertThat(primaryCaseInfoNumber).isEqualTo("X0001")
+            assertThat(caseSequence).isGreaterThan(1)
+            assertThat(beginDate).isEqualTo(LocalDate.parse("2021-01-02"))
+            assertThat(court.id).isEqualTo("LEICYC")
+            assertThat(legalCaseType.code).isEqualTo("Y")
+            assertThat(caseStatus.code).isEqualTo("I")
+            assertThat(targetCombinedCase).isNull()
+            assertThat(sourceCombinedCases).isEmpty()
+
+            // TODO use default values - no copy for now - but double check this is ok given they are not held in DPS
```

Comment:
Just not sure what to do here; data that doesn't exist in DPS but does in NOMIS - not sure if we should copy or not. I'll speak to @lbennett-moj